packages/gui/src/components/nfts/NFTCard.tsx

```diff
@@ -90,6 +100,107 @@ function NFTCard(props: NFTCardProps) {
           </CardContent>
         </CardActionArea>
       </Card>
+    );
+  }
+
+  const widthBeforeDrag = React.useRef<number | null>(null);
+  const heightBeforeDrag = React.useRef<number | null>(null);
+
+  function renderPortal() {
+    return createPortal(
+      <div
+        id="draggable-card"
+        style={{
+          position: 'absolute',
+          zIndex: '1000',
+          width: `${widthBeforeDrag.current}px`,
+          height: `${heightBeforeDrag.current}px`,
+          transform: 'scale(0.5)',
+          opacity: isHidden ? 1 : 0.6,
+        }}
+      >
+        {renderCard()}
+      </div>,
+      document.getElementById('portal') as HTMLElement
+    );
+  }
+
+  function mouseUpEvent() {
+    setTimeout(() => {
+      document.body.removeEventListener('mouseup', mouseUpEvent);
+    }, 0);
+    document.body.removeEventListener('mousemove', mouseMoveEvent);
+    setMouseDownState(false);
+    setDragState(false);
+    setDraggedNFT(null);
+  }
+
+  let startPositionLeft: number;
+  let startPositionTop: number;
+  let draggingInterval: any;
+  const countdownSpeed = 30; /* less is faster */
+  let countDown: number = countdownSpeed;
+
+  function adjustPositionBeforeAdjusted(e: any) {
+    (document.getElementById('draggable-card') as HTMLElement).style.left = `${
```

Comment:
do not use document.getElementById instead use useRef and ref in renderCard

Reply:
I don't know how to do it with createPortal :(

Reply:
I don't know how to do it with createPortal :(

Reply:
```
  function renderPortal() {
    return createPortal(
      <Box
        id="draggable-card"
        sx={{
          position: 'absolute',
          zIndex: '1000',
          width: `${widthBeforeDrag.current}px`,
          height: `${heightBeforeDrag.current}px`,
          transform: 'scale(0.5)',
          opacity: isHidden ? 1 : 0.6,
        }}
      >
        {renderCard()}
      </Box>,
      document.getElementById('portal') as HTMLElement
    );
  }
  ```

Reply:
```
  function renderPortal() {
    return createPortal(
      <Box
        id="draggable-card"
        sx={{
          position: 'absolute',
          zIndex: '1000',
          width: `${widthBeforeDrag.current}px`,
          height: `${heightBeforeDrag.current}px`,
          transform: 'scale(0.5)',
          opacity: isHidden ? 1 : 0.6,
        }}
      >
        {renderCard()}
      </Box>,
      document.getElementById('portal') as HTMLElement
    );
  }
  ```